core.md

```diff
@@ -73,11 +92,65 @@ the mechanisms needed to support that.
 
 ### Struct type definitions
 
-You can define a new type definition using the `StructType`
-constructor:
+Struct types are defined using the `StructType` constructor. There are two overloads of
+that constructor:
 
 ```js
-var PointType = new StructType({x: float64, y: float64})
+function StructType(structure, [options])
+function StructType(elementType, length, [options])
+```
+
+The first overload defines struct types with the fields given in `structure`. The
+`structure` argument must recursively consist of fields whose values are type
+definitions: either primitive or struct type definitions.
+
+The second overload is a shortcut for defining struct types with indexed elements
+of a certain type: each entry is an instance of the struct type `elementType`, and
+the length is determined by `length`.
```

Comment:
It will be important in the spec, at least, to be specific about when each version is chosen -- in particular, it ought to be keyed on the type of the first or second argument, I imagine, so that if people write a dumb forwarder like `foo(args...) { return new StructType(...args); }` (or whatever the splat syntax is) it works for either form.